src/PayU/Alu/HashService.php

```diff
@@ -1,69 +1,15 @@
 <?php
 
-namespace PayU\Alu;
 
-use PayU\Alu\Exceptions\ClientException;
+namespace PayU\Alu;
 
 /**
  * Class HashService
  * @package PayU\Alu
+ * @deprecated Will be removed on a further library version.
+ * Depending on PaymentsApi version, should use AluV3/Services/HashService or PaymentsV4/Services/HashService
  */
-class HashService
```

Comment:
If we're gonna have multiple implementations of HashService and HttpClient (v3 and v4), maybe it would be better to have an interface for each class. That way we are preparing the way for the new v4. And this would help another way. In \PayU\Alu\Client::pay we would have to use two classes we already marked as deprecated (HTTPClient, HashService) but the new interfaces.

Reply:
This was added as a future improvement for a new task.